### PR TITLE
[New Arch] Remove file

### PR DESCRIPTION
**Pull request by abelgardep**
Implements https://github.com/owncloud/android/issues/2864
Needed for https://github.com/owncloud/android/pull/3214

### CODE REVIEW

**Review by JuancaG05 — COMMENTED** [review]
LGTM, but since I'm not very familiarized with usecases, let's wait for another review

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/RemoveRemoteFileOperation.kt
Comment (davigonz) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/RemoveRemoteFileOperation.kt:60]: `{ }` even if it's just one line ?

**Review by davigonz — APPROVED** [review]

**Pull request merged**
